js/lightbeam.js

```diff
@@ -1,9 +1,63 @@
+// transforms the object of nested objects 'websites' into a
+// usable format for d3
+function transformData(websites) {
+  const nodes = [];
+  const links = [];
+  const thirdParties = [];
```

Comment:
Consider using maps, sets or objects here for thirdParties.
All provide a cleaner way to do:
```
if (thirdParties.indexOf(thirdParty) === -1) {
```

Could be:
```
if (thirdParties.has(thirdParty)) {
```

You could also instead just have:
```
function addWebsite(...) {
  // check I don't already exist yet
  // potentially merge favicons from third parties with first party nodes (Ie google.com)
  // some code
  nodes.push(node)
}
```

That way both first party and third party websites will be merged cleanly together. Currently I think you may end with a case of two nodes representing the same website.

We might also want to ditch the `"party"` key for the same reason. Google.com could be a first party and a third party to another website.